server/src/test/java/io/crate/execution/dml/upsert/UpdateToInsertTest.java

```diff
@@ -234,8 +243,38 @@ public void test_cannot_assign_to_nested_column_if_parent_is_missing() throws Ex
             e.nodeCtx,
             new CoordinatorTxnCtx(e.getSessionSettings()),
             table,
-            new String[] { "o.x" }
+            new String[] { "o.x" },
+            null
         )).isExactlyInstanceOf(UnsupportedOperationException.class)
             .hasMessage("Cannot add new child `o['x']` if parent column is missing");
     }
+
+    @Test
+    public void test_preserves_insert_column_order() throws Exception {
+        SQLExecutor e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (x int, y int, z int)")
+            .build();
+        DocTableInfo table = e.resolveTableInfo("tbl");
+        UpdateToInsert updateToInsert = new UpdateToInsert(
```

Comment:
minor: maybe add a comment of how the sql stmt looks like.

Reply:
Done